src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java

```diff
@@ -1114,7 +1122,11 @@ public boolean canCraftComplexRecipes()
         @Override
         public boolean canRecipeBeAdded()
         {
-            return canAddMoreRecipes;
+            return getMaxRecipes() > getRecipes().size() + 1;
+        }
+
+        public int getMaxRecipes() {
```

Comment:
Also formatting

Reply:
Changed formatting

Reply:
getMaxRecipes still has the curly brace in the wrong place

Reply:
Oh, there it is.